f5_openstack_agent/lbaasv2/drivers/bigip/ssl_profile.py

```diff
@@ -39,42 +38,33 @@ def create_client_ssl_profile(bigip, name, cert, key, sni_default=False):
 
         certfilename = name + '.crt'
         keyfilename = name + '.key'
-        tls_dir = tempfile.mkdtemp()
 
-        # write certificate to temp file
-        certpath = os.path.join(tls_dir, certfilename)
-        f = open(certpath, "w")
-        f.write(cert)
-        f.close()
-
-        # write key to temp file
-        keypath = os.path.join(tls_dir, keyfilename)
-        f = open(keypath, "w")
-        f.write(key)
-        f.close()
-
-        # upload files to BIG-IP
         try:
-            uploader.upload_file(str(certpath))
-            uploader.upload_file(str(keypath))
+            # In-memory upload -- data not written to local file system but
+            # is saved as a file on the BIG-IP.
+            uploader.upload_bytes(cert, certfilename)
+            uploader.upload_bytes(key, keyfilename)
+
+            # import certificate
+            param_set = {}
+            param_set['name'] = certfilename
+            param_set['from-local-file'] = \
+                '/var/config/rest/downloads/' + certfilename
```

Comment:
probably want to use the os.path.join for this type of thing.

https://docs.python.org/2/library/os.path.html